Schedulist.DAL/CSVWorkModesRepository.cs

```diff
@@ -53,35 +54,36 @@ public void AddWorkModes(WorkModesToUser workModes)
                 using (StreamWriter writer = new StreamWriter(FilePath))
                 using (var csv = new CsvWriter(writer, csvConfig))
                 {
+                    csv.WriteHeader<WorkModesToUser>();
+                    csv.NextRecord();
                     csv.WriteRecords(ListOfWorkModes);
-                    Console.Clear();
-                    Console.WriteLine($" The Workmode {workModes.WorkModeToUserID}  Has been added to the list succesfully");
+                    //Console.Clear();
+                    Console.WriteLine($" The Workmode {workModes.WorkModeName} has been added to the list succesfully at the day {workModes.dateOfWorkmode}");
+                    
                 }
             }
             catch (Exception ex)
             {
                 Console.WriteLine("An error occurred: " + ex.Message);
             }
         }
-
-        public void GetWorkModeByUserAndDate (WorkModesToUser workModes)
+        
+        public void ModifyWorkModes(WorkModesToUser workModes)
         {
+            //GetWorkModeByUserAndDate();
             var csvConfig = new CsvConfiguration(CultureInfo.InvariantCulture)
             {
                 HasHeaderRecord = true,
-                //Delimiter = "\t",
+                Delimiter = ",",
             };
             ListOfWorkModes = GetAllWorkModes();
-            try
-            {
-                ListOfWorkModes.Take(CurrentUser.currentUser.Id);
-                using (StreamReader reader = new StreamReader(FilePath));
-                //tu nie wiem, jak zrobic odczytanie przez currentusera aktualnej daty i ID usera
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine("An error occured: " +ex.Message);
-            }
+            
+        }
+        
+        public WorkModesToUser GetWorkModeByUserAndDate (int idUser, DateOnly dateWorkMode)
+        {
+            var workModesReturn = ListOfWorkModes.Where(u=>u.UserID==idUser && u.dateOfWorkmode==dateWorkMode);
```

Comment:
wotkModesReturn będzie typu IEnumerable<WorkModesToUser> (lista), a metoda zwraca typ WokModesToUser (pojedynczy obiekt). Jeśli spodziewasz się, że zwrócisz pojdynczy obiekt, to zamiast .Where() użyj .First()

Reply:
OK, tego nie wiedziałem :)